proposals/subgroups.md

```diff
@@ -0,0 +1,227 @@
+# Subgroups
+
+Status: **Draft**
+
+Last modified: 2023-11-07
+
+Issue: #4306
+
+# Requirements
+
+**Vulkan**:
+* Vulkan 1.1, and
+* subgroupSupportedStages includes compute and fragment bit, and
+* subgroupSupportedOperations includes the following bits: basic, vote, ballot, shuffle, shuffle relative, arithmetic, quad, and
+* Vulkan 1.3 or VK_EXT_subgroup_size_control
+
+According to [this query](https://vulkan.gpuinfo.org/displaycoreproperty.php?core=1.1&name=subgroupSupportedOperations&platform=all),
+~84% of devices are captured.
+Dropping quad would only grab another ~1%.
+
+**Metal**:
+* Quad-scoped permute, and
+* Simd-scoped permute, and
+* Simd-scoped reduction
+
+According to the Metal
+[feature table](https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf), this
+includes the following families: Metal3, Apple7+, Mac2.
```

Comment:
The MSL functions in this proposal have some minimum version requirements that would be worth pointing out:

- macOS: Metal 2.1.
- iOS: Metal 2.3.